tests/recipes/test_lora_finetune_single_device.py

```diff
@@ -43,19 +43,24 @@ def _get_test_config_overrides(self):
             "log_every_n_steps=1",
         ]
 
-    def _fetch_expected_loss_values(self):
-        return [10.5074, 10.5614, 10.5205, 10.4918]
+    def _fetch_expected_loss_values(self, run_qlora: bool = False):
+        if run_qlora:
+            return [10.5057, 10.5575, 10.5179, 10.4898]
+        else:
+            return [10.5074, 10.5614, 10.5205, 10.4918]
 
     @pytest.mark.integration_test
-    def test_loss(self, tmpdir, monkeypatch):
+    @pytest.mark.parametrize("run_qlora", [True, False])
+    def test_loss(self, run_qlora, tmpdir, monkeypatch):
```

Comment:
Given that the loss, config file, and overrides are all different, maybe this just merits a separate `test_qlora_loss` test case? I know it adds copypasta but I think it'd be clearer